compiler/src/dotty/tools/dotc/cc/CheckCaptures.scala

```diff
@@ -507,14 +507,29 @@ class CheckCaptures extends Recheck, SymTransformer:
       recheckFinish(result, arg, pt)
     */
 
-    /** If expected type `pt` is boxed, don't propagate free variables.
+    extension (tree: Tree)
+      private def isFunctionLiteral(using Context): Boolean = tree match
```

Comment:
TreeInfo has a `closure` extractor, which could be used instead of defining a new function here.